packages/website/src/components/lib/createEventHelper.ts

```diff
@@ -0,0 +1,19 @@
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
```

Comment:
@bradzacher referring back to #6252, what's the recommended approach here? Switching to `...args: unknown[]` causes type errors:

```plaintext
src/components/linter/createLinter.ts:28:37 - error TS2344: Type 'LinterOnParse' does not satisfy the constraint '(...args: unknown[]) => void'.
  Types of parameters 'fileName' and 'args' are incompatible.
    Type 'unknown' is not assignable to type 'string'.

28   const onParse = createEventHelper<LinterOnParse>();
                                       ~~~~~~~~~~~~~
```

Reply:
in this case we can't use Function or unknown, the only other option that can think of is

```ts
export function createEventHelper<T extends unknown[]>(): {
  trigger: (...args: T) => void;
  register: (cb: (...args: T) => void) => () => void;
} {
  const events = new Set<(...args: T) => void>();

  return {
    trigger(...args: T): void {
      events.forEach(cb => cb(...args));
    },
    register(cb: (...args: T) => void): () => void {
      events.add(cb);
      return (): void => {
        events.delete(cb);
      };
    },
  };
}

```

and use it as 

```ts
  const onLint = createEventHelper<Parameters<LinterOnLint>>();
  const onParse = createEventHelper<Parameters<LinterOnParse>>();
```

but i don't like this :)